runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -664,4 +664,44 @@ public void shutdown() {
             streamingManager.shutdown();
         }
     }
+
+    /**
+     * Populate the ProtobufDescriptorTable using the RegistryTable.
+     * Enables backward compatibility in case of data migration.
+     */
+    public void syncProtobufDescriptorTable() {
+        log.info("Running syncProtobufDescriptorTable() ...");
+
+        int numRetries = 9;
+        while (true) {
+            if (TransactionalContext.isInTransaction()) {
+                throw new IllegalThreadStateException("openTable: Called on an existing transaction");
+            }
+            try {
+                this.runtime.getObjectsView().TXBuild().type(TransactionType.WRITE_AFTER_WRITE).build().begin();
+                registryTable.forEach((tableName, corfuRecord) ->
+                        corfuRecord.getPayload().getFileDescriptorsMap()
+                                .forEach((protoName, fileDescriptorProto) -> {
+                                    ProtobufFileName fileName = ProtobufFileName
+                                            .newBuilder().setFileName(protoName).build();
+                                    ProtobufFileDescriptor fileDescriptor = ProtobufFileDescriptor
+                                            .newBuilder().setFileDescriptor(fileDescriptorProto).build();
+                                    protobufDescriptorTable.putIfAbsent(fileName, new CorfuRecord<>(
+                                            fileDescriptor, corfuRecord.getMetadata()));
+                                }));
+                this.runtime.getObjectsView().TXEnd();
+                log.info("syncProtobufDescriptorTable: completed!");
```

Comment:
is it worth to add info about # of descriptors added and tables? (for debugging) for instance, completed sync of X proto descriptors corresponding to Z tables.